src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/UnsupportedMarshallingFactory.cs

```diff
@@ -12,7 +15,34 @@ public sealed class UnsupportedMarshallingFactory : IMarshallingGeneratorFactory
             TypePositionInfo info,
             StubCodeContext context)
         {
+            if (info.MarshallingAttributeInfo is NoMarshallingInfo && CustomTypeToErrorMessageMap.TryGetValue(info.ManagedType, out string errorMessage))
+            {
+                throw new MarshallingNotSupportedException(info, context)
+                {
+                    NotSupportedDetails = errorMessage
+                };
+            }
             throw new MarshallingNotSupportedException(info, context);
         }
+
+        public UnsupportedMarshallingFactory()
+            :this(DefaultTypeToErrorMessageMap)
+        {
+
+        }
+
+        public UnsupportedMarshallingFactory(ImmutableDictionary<ManagedTypeInfo, string> customTypeToErrorMessageMap)
```

Comment:
```suggestion
        private UnsupportedMarshallingFactory(ImmutableDictionary<ManagedTypeInfo, string> customTypeToErrorMessageMap)
```

At least until we see a use case for it.

Reply:
I couldn't find one in this review.